python/ray/tune/tests/test_tuner.py

```diff
@@ -350,38 +353,40 @@ def train_func(config):
         trial_dir = Path(session.get_trial_dir())
         with open(trial_dir / "write.txt", "w") as f:
             f.write(f"{config['id']}")
-        # Make sure we didn't write to the working dir
-        assert not os.path.exists(orig_working_dir / "write.txt")
-        # Make sure that the file we wrote to isn't overwritten
-        assert open(trial_dir / "write.txt", "r").read() == f"{config['id']}"
 
     tuner = Tuner(
         train_func,
-        tune_config=TuneConfig(
-            chdir_to_trial_dir=False,
-        ),
+        tune_config=TuneConfig(chdir_to_trial_dir=False),
         param_space={"id": tune.grid_search(list(range(4)))},
     )
-    tuner.fit()
+    results = tuner.fit()
+    assert not results.errors
+    for result in results:
+        artifact_data = open(result.log_dir / "write.txt", "r").read()
+        assert artifact_data == f"{result.config['id']}"
+
+    os.chdir(old_cwd)
     ray.shutdown()
 
 
 @pytest.mark.parametrize("runtime_env", [{}, {"working_dir": "."}])
-def test_tuner_relative_pathing_with_env_vars(runtime_env):
+def test_tuner_relative_pathing_with_env_vars(runtime_env, tmpdir):
     """Tests that `TUNE_ORIG_WORKING_DIR` environment variable can be used to access
     relative paths to the original working directory.
     """
+    old_cwd = os.getcwd()
+    os.chdir(tmpdir)
```

Comment:
This is a nit, but we should probably introduce a fixture here, e.g. `chdir_tmpdir`. This is to ensure we also chdir back to the old wd if the test fails. Otherwise we could run into test problems in subsequent tests if the current working dir does not exist (because the tmpdir was deleted)

Something like

```
@pytest.fixture
def chdir_tmpdir(tmpdir):
    old_cwd = os.getcwd()
    os.chdir(tmpdir)
    yield tmpdir
    os.chdir(old_cwd)
```

should work